specs/operator/pre-upgrade.spec

```diff
@@ -0,0 +1,98 @@
+  * Create & Link secret "github-secret" to service account "pipeline"
+  * Expose Event listener "listener-ctb-github-push"
+  * Mock post event to "github" interceptor with event-type "push", payload "testdata/triggers/github-ctb/push.json", with TLS "false"
+  * Assert eventlistener response
+  * Verify pipelinerun
+    |S.NO|pipeline_run_name       |status    |check_label_propagation|
+    |----|------------------------|----------|-----------------------|
+    |1   |pipelinerun-git-push-ctb|successful|no                     |
+  * Delete resource "pipelinerun-git-push-ctb" of type "pipelinerun"
+  * Create
+    |S.NO|resource_dir                                               |
+    |----|-----------------------------------------------------------|
+    |1   |testdata/triggers/triggersCRD/eventlistener-triggerref.yaml|
+    |2   |testdata/triggers/triggersCRD/trigger.yaml                 |
+    |3   |testdata/triggers/triggersCRD/triggerbindings.yaml         |
+    |4   |testdata/triggers/triggersCRD/triggertemplate.yaml         |
+    |5   |testdata/triggers/triggersCRD/pipeline.yaml                |
+  * Expose Event listener "listener-triggerref"
+  * Mock post event to "github" interceptor with event-type "pull_request", payload "testdata/triggers/triggersCRD/pull-request.json", with TLS "false"
+  * Assert eventlistener response
+  * Verify pipelinerun
+    |S.NO|pipeline_run_name       |status    |check_label_propagation|
+    |----|------------------------|----------|-----------------------|
+    |1   |parallel-pipelinerun    |successful|no                     |
+  * Delete resource "parallel-pipelinerun" of type "pipelinerun"
+  * Create
+    |S.NO|resource_dir                                                        |
+    |----|--------------------------------------------------------------------|
+    |1   |testdata/triggers/bitbucket/bitbucket-eventlistener-interceptor.yaml|
+  * Create & Link secret "bitbucket-secret" to service account "pipeline"
+  * Expose Event listener "bitbucket-listener"
+  * Mock post event to "bitbucket" interceptor with event-type "refs_changed", payload "testdata/triggers/bitbucket/refs-change-event.json", with TLS "false"
+  * Assert eventlistener response
+  * Verify taskrun
+    |S.NO|task_run_name    |status |
+    |----|-----------------|-------|
+    |1   |bitbucket-run    |Failure|
+  * Delete resource "bitbucket-run" of type "taskrun"
+
+## Setup S2I nodejs pipeline pre upgrade: PIPELINES-18-TC02
+Tags: pre-upgrade, e2e, clustertasks, non-admin, s2i
+Component: Pipelines
+Level: Integration
+Type: Functional
+Importance: Critical
+
+Steps:
+  * Create project "releasetest-upgrade-s2i"
+  * Create
+      |S.NO|resource_dir                                          |
+      |----|------------------------------------------------------|
+      |1   |testdata/v1beta1/clustertask/pipelines/s2i-nodejs.yaml|
+      |2   |testdata/v1beta1/clustertask/pvc/pvc.yaml             |
```

Comment:
What's the purpose of these two resources?

Reply:
Before the upgrade, I am creating S2I pipeline and after upgrade, I am just verifying that the pipeline works fine. If there is any change in the s2i pipeline/clustertask we can catch it here.